ee_plugin/utils.py

```diff
@@ -4,178 +4,256 @@
 """
 
 import json
+import tempfile
 
 import ee
 import qgis
-from qgis.core import (
-    QgsProject,
-    QgsRasterLayer,
-)
-
+from qgis.core import QgsProject, QgsRasterLayer, QgsVectorLayer
 from .ee_plugin import VERSION as ee_plugin_version
 
 
+def is_named_dataset(eeObject):
+    """
+    Checks if the FeatureCollection is a named dataset that should be handled as a vector tiled layer.
+    """
+    try:
+        table_id = eeObject.args.get("tableId", "")
+        return bool(table_id)  # If tableId exists, it's a named dataset
+    except AttributeError:
+        return False
+
+
+def get_layer_by_name(name):
+    layers = QgsProject.instance().mapLayers().values()
+
+    for layer in layers:
+        if layer.name() == name:
+            return layer
+
+    return None
```

Comment:
Should we use `mapLayersByName()`? https://qgis.org/pyqgis/3.40/core/QgsProject.html#qgis.core.QgsProject.mapLayersByName

```suggestion
    for layer in QgsProject.instance().mapLayersByName(name):
        return layer
```

Also, being that a list of layers is returned, I assume that layer names are not guaranteed to be unique.  Not sure if that will cause issues someday.

Reply:
great suggestion! added.